packages/amesos2/src/Amesos2_TpetraMultiVecAdapter_def.hpp

```diff
@@ -211,23 +211,14 @@ namespace Amesos2 {
           const size_t lclNumRows = redist_mv.getLocalLength();
           for (size_t j = 0; j < redist_mv.getNumVectors(); ++j) {
             auto av_j = av(lda*j, lclNumRows);
-            auto X_j = redist_mv.getVector(j);
+            //auto X_j = redist_mv.getVector(j);
             auto X_lcl_j_2d = redist_mv.template getLocalView<host_execution_space> ();
             auto X_lcl_j_1d = Kokkos::subview (X_lcl_j_2d, Kokkos::ALL (), j);
             for ( size_t i = 0; i < lclNumRows; ++i ) {
               av_j[i] = X_lcl_j_1d(i);
```

Comment:
This is broken with respect to UVM -- you'll need a fence before accessing these data on host.  That can be a separate issue, though.

Reply:
Good catch, I'll fix it now otherwise it may end up in a list of things that keeps growing.